_usage/hint_mapping.md

```diff
@@ -0,0 +1,295 @@
+
+Modify your PATH environment variable to include the target PostgreSQL installation.
+
+Modify the `postgresql.conf` file, adding `pg_hint_plan` to the `shared_preload_libraries` or `session_preload_libraries` parameter. Then you can use pg_hint_plan’s query hints in T-SQL connections.
+
+
+### Using T-SQL Hint Mapping with Babelfish
+
+By default, Babelfish ignores all T-SQL hints. To apply a T-SQL hint, you need to turn on the `babelfishpg_tsql.enable_hint_mapping` GUC; you can use the following command on the Babelfish TDS port:
+
+```sql
+SELECT set_config('babelfishpg_tsql.enable_hint_mapping', 'on', false);
+```
+
+### Supported Hints
+
+After setting `babelfishpg_tsql.enable_hint_mapping` to `ON` and loading `pg_hint_plan` you can use the following [T-SQL hints](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-query?view=sql-server-ver16) on the TDS port.
+
+#### Index Hints
+
+You can specify an [INDEX hint](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-join?view=sql-server-ver16) in two ways:
+
+- As a table hint; for example:
+
+```sql
+SELECT * FROM t1 (index(index_t1_b1)) WHERE b1 = 1
+SELECT * FROM t1 with(index(index_t1_b1)) WHERE b1 = 1
+```
+
+- As a query hint through the OPTION clause; for example:
+
+SELECT * FROM t1 WHERE b1=1 OPTION(table hint(t1, index(index_t1_b1)))
+
+#### Join Hint
+
+You can specify a [JOIN hint](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-join?view=sql-server-ver16) in two ways:
+
+- As a join hint; for example:
+
+```sql
+SELECT * FROM t1 INNER MERGE JOIN t2 ON t1.a1 = t2.a2
+```
+
+- As a query hint; for example:
+
+```sql
+SELECT * FROM t1 JOIN t2 ON t1.a1 = t2.a2 OPTION(loop join)
+```
+
+A query hint may include a join hint specified both as a join hint and a query hint in a conflicting manner. An example of a conflicting join hint is:
+
+```sql
+SELECT * FROM t1 INNER MERGE JOIN t2 ON t1.a1 = t2.a2 OPTION(HASH JOIN)
+```
+
+In this case, the server returns the following error: C
+
+```sql
+Conflicting JOIN optimizer hints specified
+```
+
+#### FORCE ORDER Hint
+
+You can use a [FORCE ORDER hint](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-join?view=sql-server-ver16) as a query hint specified with the OPTION clause. A force order hint specifies that the join order indicated by the query syntax is preserved during query optimization; for example:
+
+```sql
+SELECT * FROM t1 
+            JOIN t2 ON t1.a1 = t2.a2
+            JOIN t3 ON t1.a1 = t3.a3 OPTION(FORCE ORDER)
+```
+
+#### MAXDOP Hint
+
+You can use a [MAXDOP hint](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-join?view=sql-server-ver16) as a query hint specified with the OPTION clause. To apply this hint, you must first set the PostgreSQL `max_parallel_workers_per_gather` GUC to the value specified in the hint; for example:
```

Comment:
Customers don't need to set the GUC max_parallel_workers_per_gather. We do that as part of the implementation. Also, it would be good to mention:
If the value specified is greater than the the actual available processors at run time, the query optimizer can only use the maximum available processors. If the hint value of 0 is specified, the query optimizer tries to use all the available processors up to 64 processors.